auto-value-moshi/src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtension.java

```diff
@@ -377,14 +378,19 @@ public MethodSpec createWriteMethod(TypeName autoValueClassName,
         .addException(IOException.class);
 
     writeMethod.addStatement("$N.beginObject()", writer);
+
+    NameAllocator nameAllocator = new NameAllocator();
```

Comment:
Also you either need to add the adapter names or always qualify them with `this.`. Please add a test with two properties, "foo" and "fooAdapter" to ensure this works.